modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -683,6 +678,28 @@ private void _addObjectDefinitionResources(
 		}
 
 		if (objectRelationships != null) {
+			ArrayList<String> objectRelationshipsNames = new ArrayList<>();
+
+			for (ObjectRelationship objectRelationship : objectRelationships) {
+				objectRelationshipsNames.add(objectRelationship.getName());
+			}
+
+			List<com.liferay.object.model.ObjectRelationship>
+				oldObjectRelationships =
+					_objectRelationshipLocalService.getObjectRelationships(
+						QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+			for (com.liferay.object.model.ObjectRelationship
```

Comment:
To avoid this situation that I mentioned above, I believe it can be used get method to bring all the relationships belonging only to the current objectDefinition

To buy space, the list can be passed inline in the **for** statament like this:

```java
for (com.liferay.object.model.ObjectRelationship
	oldObjectRelationship :
		_objectRelationshipLocalService.getObjectRelationships(
			objectDefinitionId)) {
      ...
 }
```

Additionaly, I have a suggestion, following your approach, this same **for** can be used to update the existing relationships and make more easy the add of the new ones. With that, do all the needed operations in only one **for**:

*is more easy to track **this** = *is more easy to track **these**


![suggestion](https://github.com/liferay-objects/liferay-portal/assets/79872134/bb1dd0cc-151d-4928-8575-f1b196b4f5ab)

Reply:
I believe that my suggestion save more lines and, maybe, be more clear.
But I don't know if become more easy to read... or more hard. Feel free to follow, or not.

Reply:
Thanks for the tip, @p-albuquerque, it was a great catch!
I will not apply the `HashMap` approach because I'm already iterating over the `objectRelationships` argument, then it is not necessary to come up with a new data structure containing the same data.